api/src/routes/plan-route.js

```diff
@@ -0,0 +1,208 @@
+ *        content:
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/Plan'
+ *      responses:
+ *        "200":
+ *          description: A plan schema
+ *          content:
+ *            application/json:
+ *              schema:
+ *                $ref: '#/components/schemas/Plan'
+ *        "400":
+ *          description: Database error
+ */
+router.put("/", async (req, res) => {
+    Plan.findOneAndUpdate({ name: req.body.name }, req.body, { upsert: "true" }, (err, plan) => {
+        if (err) {
+            LOGGER.error(err);
+            res.status(400).json({ msg: err.message });
+        } else {
+            //change to send the updated plan, rather than the old plan
+            Plan.findById(plan._id, (err, updatedPlan) => {
+                LOGGER.info("PUT request succeeded for /Plan/");
+                res.status(200).json(updatedPlan);
+            });
+        }
+    });
+});
+
+// Creates a new plan
+/**
+ * @swagger
+ * path:
+ *  /plan/:
+ *    post:
+ *      tags: [Plan]
+ *      summary: Creates a Plan
+ *      requestBody:
+ *        required: true
+ *        content:
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/Plan'
+ *      responses:
+ *        "201":
+ *          description: A plan object is created
+ *        "400":
+ *          description: Database error
+ */
+router.post("/", async (req, res) => {
+    const newPlan = new Plan(req.body);
+    newPlan.save((err, plan) => {
+        if (err) {
+            LOGGER.error(err);
+            res.status(400).json({ msg: err.message });
+        } else {
+            LOGGER.info("POST request succeeded for /Plan/");
+            LOGGER.info(plan);
+            res.status(201).json(plan);
+        }
+    });
+});
+
+// deletes a plan
+/**
+ * @swagger
+ * path:
+ *  /plan/{name}:
+ *    delete:
+ *      tags: [Plan]
+ *      summary: Deletes a Plan
+ *      parameters:
+ *        - in: path
+ *          name: name
+ *          schema:
+ *            type: string
+ *          required: true
+ *          description: plan code of plan
+ *      responses:
+ *        "200":
+ *          description: A plan schema
+ *          content:
+ *            application/json:
+ *              schema:
+ *                $ref: '#/components/schemas/Plan'
+ *        "404":
+ *          description: No plan found
+ *        "400":
+ *          description: Database error (internal)
+ */
+router.delete("/:name", async (req, res) => {
+    Plan.findOneAndDelete({ name: req.params.name }, (err, plan) => {
+        if (err) {
+            LOGGER.error(err);
+            res.status(400).json({ msg: err.message });
+        } else {
+            if (plan === null) {
+                LOGGER.error("Plan does not exist");
+                res.status(404).json({ msg: "Requested object not found" });
+            } else {
+                LOGGER.info("DELETE request succeeded for /Plan/:name");
```

Comment:
change :name to actually send the name so do something like `${req.params.name}`

Reply:
Fixed